lib/routemaster/redis_broker.rb

```diff
@@ -25,18 +27,42 @@ def cleanup
       _cleanup
     end
 
+    # Allow to inject pre-built Redis clients
+    #
+    # Before storing a new connection, ensure that any previously
+    # set client is properly closed.
+    #
+    def inject(clients={})
```

Comment:
Makes sense, I guess how I would expect it to work would be to inject an instance of `RedisBroker` so make it not a `Singleton` but an actual instance, which one accesses via `Config` and then inject this. But I can see this is significantly more work so I would say your call really. Just wanted to be aware of it really.